### PR TITLE
feat: make sure there's no empty header in light-client creation mode

### PR DESCRIPTION
**Pull request by ashuralyk**
bug: make default header with specified slot while getting error from get_header rpc

chore: adjust log-printing in the Forcerelay spawning task

Closes: #XXX

## Description
@yangby-cryptape please make sure the `get_header` fallback logic is proper, and after merging your previous PR which can be enabled even if headers are discontinuous, the header-aligner will remove empty headers from the left side in the `update_headers`

______

### PR author checklist:
- [ ] Added changelog entry, using [`unclog`](https://github.com/informalsystems/unclog).
- [ ] Added tests: integration (for Hermes) or unit/mock tests (for modules).
- [ ] Linked to GitHub issue.
- [ ] Updated code comments and documentation (e.g., `docs/`).
- [ ] Tagged *one* reviewer who will be the one responsible for shepherding this PR.

### Reviewer checklist:

- [ ] Reviewed `Files changed` in the GitHub PR explorer.
- [ ] Manually tested (in case integration/unit/mock tests are absent).

### CODE REVIEW

File: crates/relayer/src/chain/ckb/utils.rs [hunk 1]
```diff
@@ -366,6 +372,13 @@ mod tests {
         let path = TempDir::new().unwrap();
         let storage: Storage<MainnetEthSpec> = Storage::new(path).unwrap();
 
+        (chain_id, updates_part_1, updates_part_2, storage)
+    }
```
Comment (chaoticlonghair) [inline crates/relayer/src/chain/ckb/utils.rs]: Leave this closure, the destructor of [`tempfile::TempDir`] will run automatically in [its `Drop` implement].

Then, the files of storage could be broken.

So, the result of the unit tests will be randomized.

A very important tip for using crate `tempfile` is that to make sure the lifetime of [`tempfile::TempDir`] is longer than the lifetime of structs which use it.

References:
- [`tempfile::TempDir`]
- [`MockStore` in CKB](https://github.com/nervosnetwork/ckb/blob/develop/util/test-chain-utils/src/mock_store.rs#L20-L27) (the comment in this piece code is useful)


p.s. I would like to push more commits base on this PR, so you don't have to update this issue, I will do that.

p.s.s. Ref: #101

 [`tempfile::TempDir`]: https://docs.rs/tempfile/latest/tempfile/struct.TempDir.html
[its `Drop` implement]: https://docs.rs/tempfile/latest/tempfile/struct.TempDir.html#impl-Drop

Comment (ashuralyk) [reply crates/relayer/src/chain/ckb/utils.rs]: ok, it's fixed with your previous PR.

File: crates/relayer-types/src/clients/ics07_eth/types.rs [hunk 2]
```diff
@@ -103,6 +103,14 @@ impl Update {
             ..Default::default()
         }
     }
+
+    pub fn is_finalized_empty(&self) -> bool {
+        let empty_header = Header {
+            slot: self.finalized_header.slot,
+            ..Default::default()
+        };
+        empty_header == self.finalized_header
```
Comment (chaoticlonghair) [inline crates/relayer-types/src/clients/ics07_eth/types.rs]: Unnecessary construction.

File: crates/relayer/src/chain/ckb/utils.rs [hunk 3]
```diff
@@ -97,6 +97,25 @@ where
...
+        for header in header_updates.iter() {
+            if header.is_finalized_empty() {
+                trim_count += 1;
+            } else {
+                break;
+            }
+        }
+        (0..trim_count).for_each(|_| {
+            header_updates.remove(0);
+        });
```
Comment (chaoticlonghair) [inline crates/relayer/src/chain/ckb/utils.rs]: [`fn skip_while<P>(self, predicate: P) -> ...`](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.skip_while)

File: crates/relayer/src/chain/ckb/utils.rs [hunk 4]
```diff
@@ -85,9 +85,9 @@ where
 
 pub fn align_native_and_onchain_updates<S, E>(
     chain_id: &str,
-    header_updates: &Vec<EthUpdate>,
+    header_updates: &mut Vec<EthUpdate>,
```
Comment (chaoticlonghair) [inline crates/relayer/src/chain/ckb/utils.rs:88]: If I remembered correctly, `clippy` will throw errors.
- `&Vec<T>` -> `&[T]`
- `&mut Vec<T>` -> `&mut [T]`

Comment (ashuralyk) [reply crates/relayer/src/chain/ckb/utils.rs:88]: no error throw, maybe clippy could pass for the mutable use

File: crates/relayer-types/src/clients/ics07_eth/types.rs [hunk 5]
```diff
@@ -103,6 +103,15 @@ impl Update {
...
         }
     }
+
+    pub fn is_finalized_empty(&self) -> bool {
+        let header = &self.finalized_header;
+        header.slot > 0
+            && header.proposer_index == 0
+            && header.parent_root == Default::default()
+            && header.state_root == Default::default()
+            && header.body_root == Default::default()
```
Comment (chaoticlonghair) [inline crates/relayer-types/src/clients/ics07_eth/types.rs:113]: Unnecessary construction * 3.

**Review by chaoticlonghair — APPROVED** [review]
I approved because the code logic looks good to me; but the coding is not good enough.

File: crates/relayer-types/src/clients/ics07_eth/types.rs (hunk 5 above)
Comment (ashuralyk) [reply crates/relayer-types/src/clients/ics07_eth/types.rs:113]: ok, you can change these later when submitting new PRs.

**Pull request merged**